### ISSUE TITLE
Question about valid_batch_data

### ISSUE THREAD

**Issue by lan2720** [tobegit3hub/tensorflow_template_application#5]
Hi tobe,

```
# Read TFRecords file for validatioin
validate_filename_queue = tf.train.string_input_producer(
    tf.train.match_filenames_once("data/cancer_test.csv.tfrecords"),
    num_epochs=epoch_number)
validate_label, validate_features = read_and_decode(validate_filename_queue)
validate_batch_labels, validate_batch_features = tf.train.shuffle_batch(
    [validate_label, validate_features],
    batch_size=validate_batch_size,
    num_threads=thread_number,
    capacity=capacity,
    min_after_dequeue=min_after_dequeue)
```

In the code, the validation batch data are also generated by QueueRunner. So cooridinator will collect these valid batch threads together with train QueueRunner right?

But if I have a different batch\_size settings in train dataset and valid dataset. Their total num are also different( most of time). What should the coordiantor work? 

And I found you set the `num_of_epochs` the same value in train and valid queuerunner. Why it like that?
I think the two num_epoch should not be equal?
```

flags.DEFINE_integer('epoch_number', None, 'Number of epochs to run trainer.')

filename_queue = tf.train.string_input_producer(
    tf.train.match_filenames_once("data/cancer_train.csv.tfrecords"),
    num_epochs=epoch_number)

validate_filename_queue = tf.train.string_input_producer(
    tf.train.match_filenames_once("data/cancer_test.csv.tfrecords"),
    num_epochs=epoch_number)
```

Thanks!

**Comment by lan2720** [comment]
Another question, 

```
flags.DEFINE_integer("batch_size", 1024,
                     "indicates batch size in a single gpu, default is 1024")
flags.DEFINE_integer("validate_batch_size", 1024,
                     "indicates batch size in a single gpu, default is 1024")
```

Train.csv and test.csv both contain several hundred examples. Could batch_size be 1024 here, which is bigger than the total num of dataset? I'm confused.

**Comment by tobegit3hub** [comment]
Sorry for the late reply.

Actually the batch sizes of training and validate are not required to be equal. You can set different values for them and we use the same parameter for simplicity. 

The batch size is allowed to be more than the number of lines in training data. TensorFlow will shuffle batch to get data from files in many times.

**Comment by tobegit3hub** [comment]
I'm gonna close this if you don't have any question. @lan2720 

Feel free to re-open this if needed 😃
